api/pkg/shared/resource/resource.go

```diff
@@ -314,6 +316,19 @@ func filterResourceName(match, name string) func(db *gorm.DB) *gorm.DB {
 	}
 }
 
+func filterByCatalogs(catalogs []string) func(db *gorm.DB) *gorm.DB {
+	if len(catalogs) == 0 {
+		return noop
+	}
+
+	catalogs = lower(catalogs)
+	return func(db *gorm.DB) *gorm.DB {
+		return db.Model(&model.Resource{}).
+			Joins("JOIN catalogs as ct on ct.id = resources.catalog_id").
```

Comment:
I am not sure if we require `join` for this, we already `Preload` catalog data so we can just add a where statement. wdyt?